xray/src/generation.rs

```diff
@@ -227,14 +257,22 @@ impl ColoringStrategy for IntensityColoringStrategy {
                 {
                     Entry::Occupied(mut e) => {
                         let per_column_data = e.get_mut();
-                        per_column_data.sum += intensity;
-                        per_column_data.count += 1;
+                        let data = per_column_data
+                            .entry(bins[i])
+                            .or_insert(IntensityPerColumnData { sum: 0.0, count: 0 });
```

Comment:
nit: do you think it would make sense to derive the `Default` trait for `IntensityPerColumnData` and use `or_default` here?

Reply:
Done.